File: ios/RNTStripeCardInputManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(RNTStripeCardInputManager, RCTViewManager)

RCT_EXTERN_METHOD(focus:(nonnull NSNumber*) reactTag)
RCT_EXTERN_METHOD(blur:(nonnull NSNumber*) reactTag)
RCT_EXTERN_METHOD(clear:(nonnull NSNumber*) reactTag)

RCT_EXPORT_VIEW_PROPERTY(onCardValidCallback, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onFocusChange, RCTBubblingEventBlock)

RCT_EXPORT_VIEW_PROPERTY(postalCodeEntryEnabled, BOOL)
RCT_EXPORT_VIEW_PROPERTY(textColor, UIColor)
RCT_EXPORT_VIEW_PROPERTY(placeholderColor, UIColor)
RCT_EXPORT_VIEW_PROPERTY(borderColor, UIColor)
RCT_EXPORT_VIEW_PROPERTY(borderWidth, CGFloat)
RCT_REMAP_VIEW_PROPERTY(backgroundColor, cardBackgroundColor, UIColor)
RCT_EXPORT_VIEW_PROPERTY(enabled, BOOL)

@end

File: ios/RNTStripeCardInputManager.swift
import Foundation
import Stripe

@objc(RNTStripeCardInputManager)
class RNTStripeCardInputManager: RCTViewManager {
    override func view() -> UIView! {
        return RNTStripeCardInputView()
    }
    
    @objc func focus(_ reactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (uiManager: RCTUIManager?, viewRegistry:[NSNumber : UIView]?) in
            let view: RNTStripeCardInputView = viewRegistry![reactTag] as! RNTStripeCardInputView;
            view.focus()
          }
    }
    
    @objc func blur(_ reactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (uiManager: RCTUIManager?, viewRegistry:[NSNumber : UIView]?) in
            let view: RNTStripeCardInputView = viewRegistry![reactTag] as! RNTStripeCardInputView;
            view.blur()
          }
    }
    
    @objc func clear(_ reactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (uiManager: RCTUIManager?, viewRegistry:[NSNumber : UIView]?) in
            let view: RNTStripeCardInputView = viewRegistry![reactTag] as! RNTStripeCardInputView;
            view.clear()
          }
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
